packages/matter.js/src/device/AutoDevice.ts

```diff
@@ -0,0 +1,361 @@
+type UntypedClusterServer = <() => >;
+
+/**
+ * Construction options for AutoDevice.
+ */
+export type AutoDeviceOptions<State> = {
+    state?: Partial<State>;
+    endpointId?: number;
+}
+
+// Automatic wiring support class
+class WiringContext {
+    private serverHandlers: UntypedHandlers = {};
+
+    constructor(
+        // The cluster to wire
+        public cluster: ClusterModel,
+
+        // Hooks invoked at construction time to do perform setup
+        public initializers: Array<(this: AutoDevice, options: AutoDeviceOptions<BaseState>) => void>,
+
+        // Target prototype for device class
+        public deviceProto: any,
+
+        // Target prototype for state class
+        public stateProto: any,
+    ) {}
+
+    // Auto-wire state and device interfaces for a specific cluster
+    wire() {
+        this.cluster.attributes.forEach((attr) => {
+            this.wireAttributeState(attr);
+        });
+        this.wireAttributeEvents();
+
+        this.cluster.commands.forEach((command) => {
+            this.wireCommand(command);
+        });
+        
+        this.cluster.events.forEach((event) => {
+            this.wireEvent(event);
+        });
+
+        this.wireClusterServer();
+    }
+
+    // Implement the state interface
+    private wireAttributeState(attr: AttributeModel) {
+        const cluster = this.cluster;
+
+        // Getter
+        const descriptor = <PropertyDescriptor>{
+            get(this: BaseState) {
+                return this[DEVICE].attributeFor(cluster, attr).get();
+            }
+        };
+
+        // Setter
+        if (attr.fixed) {
+            descriptor.set = function(this: BaseState, value) {
+                this[DEVICE].attributeFor(cluster, attr).setLocal(value);
+                return true;
+            }
+        } else {
+            descriptor.writable = false;
+        }
+
+        // Add the property
+        Object.defineProperty(this.stateProto, attr.name, descriptor);
+
+        // If there's a default "get" handler, install it
+        const handler = cluster.defaultServerHandlers[attr.getter];
+        if (handler) this.serverHandlers[attr.getter] = handler;
+    }
+
+    // Implement attribute change events
+    private wireAttributeEvents() {
+        const cluster = this.cluster;
+
+        this.initializers.push(function(this: AutoDevice) {
+            cluster.attributes.forEach((attr) => {
+                const clusterAttr = this.attributeFor(cluster, attr);
+                clusterAttr.addListener((...args) =>
+                    (<any>this)[attr.handler](...args));
+            });
+        });
+    }
+
+    // Implement device commands
+    private wireCommand(command: CommandModel) {
+        this.deviceProto[command.invoker] = ignore;
+        this.serverHandlers[command.handler] =
+            async (request: any, _session, _message, endpoint: any) =>
+                await endpoint[command.handler](request);
```

Comment:
see above - this ill not work that way unless we change the whole InteractionServer logic. Commands are triggered from remote, so the Device class normally plays no role into it. The only thing is to make sure that the registered ClusterServer handlers know that there is a better implementation "upstairs" :-)

Reply:
PS: Ok in this.defaultServerHandlers you know the default ones, but they are not considered here

Reply:
D'oh I totally botched that implementation.

Reply:
OK I pushed the implementation that I intended.  (I think, still untested.)  See the new wireCommand

I also forgot to add the empty callback for attribute change events.  That's in wireAttributeState